views/profile_view/views.py

```diff
@@ -40,11 +40,11 @@ def get(self, request, username=None):
         except Http404:
             return redirect('home')
         context = {
-            'request_username': f"{request.user.first_name} {request.user.last_name}",
+            'full_name': f"{request.user.first_name} {request.user.last_name}",
             'user_profile': user_profile,
             'isAdmin': request.user.role == 'Admin',
             'self': username == None,
-            'username': username,
+            'username': "" if username is None else username,
```

Comment:
we can do request.user.username if the username is None because it's the own users profile when there is no username.